src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { configureTestBed } from '~/testing/unit-test-helper';
+import { AuthStorageDirective } from './auth-storage.directive';
+@Component({
+  template: `<div *cdAuthStorage="condition"></div>`
+})
+export class AuthStorageDirectiveTestComponent {
+  condition: string | string[] | object;
+}
+
+describe('AuthStorageDirective', () => {
+  let fixture: ComponentFixture<AuthStorageDirectiveTestComponent>;
+  let component: AuthStorageDirectiveTestComponent;
+  let getPermissionsSpy: jasmine.Spy;
+  const getDiv = () => fixture.debugElement.nativeElement.querySelector('div');
+
+  configureTestBed({
+    declarations: [AuthStorageDirective, AuthStorageDirectiveTestComponent],
+    providers: [AuthStorageService]
+  });
+
+  beforeEach(() => {
+    getPermissionsSpy = spyOn(TestBed.inject(AuthStorageService), 'getPermissions');
+    getPermissionsSpy.and.returnValue(new Permissions({ osd: ['read'], rgw: ['read'] }));
+    fixture = TestBed.createComponent(AuthStorageDirectiveTestComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
```

Comment:
It seems this line is not needed (it was removed in the suggestion I made).

Reply:
Oh! I must have missed it. I'll remove